components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +132,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.where(column, OperatorType.LIKE, "%");
```

Comment:
@LosD & CHAR(5) '' => '_____': I would think so too. It might be database/editor/server configuration issue but it behaves as I described (I've just rechecked).